---
title: "Rarefaction"
author: "Haydeé"
date: "`r Sys.Date()`"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Cargamos librerías y datos de las dos minas Pmt y S1.

```{r}
library("phyloseq")
library("ggplot2")
library("RColorBrewer")
library("patchwork")
library(vegan)
setwd("~/metagenomas")


raw_metagenomes <- import_biom("metagenomes_Pmt_S1.biom")
raw_metagenomes@tax_table@.Data <- substring(raw_metagenomes@tax_table@.Data, 4)
colnames(raw_metagenomes@tax_table@.Data)<- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
unique(raw_metagenomes@tax_table@.Data[,"Kingdom"])
raw_metagenomes <- subset_taxa(raw_metagenomes, Kingdom == "Bacteria")
```

Extraemos los OTUs y calculamos el número máximo de conteos.

```{r}
otu <- otu_table(raw_metagenomes)
otu <- as.data.frame(t(otu))
sample_names <- rownames(otu)

S <- specnumber(otu)
S
raremax <- min(rowSums(otu))
raremax
```

Aplicamos rarefy con los OTUs, número máximo de conteos y el paso para aplicar el rarefy.

```{r}
rare1 <- rarefy(t(otu),  sample= c(seq(100000,2000000 , by = 100000), raremax)  , se = T, MARG = 2)
rare_t <- t(rare1)
```

Guardamos los datos en una base de datos, calculamos las desviaciones estándar.
```{r}
site <- c(rep("Pmt2_S287",21), rep("S1_S286",21))

reads <- c(c(seq(100000,2000000 , by = 100000), raremax), c(seq(100000,2000000 , by = 100000), raremax)) 

Pmt2_S <- rare_t[,1]
Pmt2_se <- rare_t[,2]
S1_S <- rare_t[,3]
S1_se <- rare_t[,4]

Svalues <- c(Pmt2_S, S1_S)
sevalues <- c(Pmt2_se, S1_se)

data <- data.frame(Site= site, 
                   reads = reads,
                   S = Svalues,
                   se = sevalues)
```


Realizamos el plot.
```{r}
rare_plot <- ggplot(data, aes(x= reads, y=S, col=Site)) +
  geom_point()+
  geom_errorbar(aes(ymin=S-se, ymax=S+se), width=.3,
                position=position_dodge(0.001))+
  ylab("OTUs")

rare_plot
```